From 5a216abaa737195327235e37563b18a6bf2a74dc Mon Sep 17 00:00:00 2001
From: Philippe Mathieu-Daude <philmd@redhat.com>
Date: Thu, 1 Aug 2019 20:43:47 +0200
Subject: [PATCH 1/3] ArmVirtPkg: silence DEBUG_VERBOSE masking ~0x00400000 in
 QemuRamfbDxe (RH only)

Message-id: <20190801184349.28512-2-philmd@redhat.com>
Patchwork-id: 89859
O-Subject:  [RHEL-8.1.0 edk2 PATCH v4 1/3] ArmVirtPkg: silence DEBUG_VERBOSE
	masking ~0x00400000 in QemuRamfbDxe (RH only)
Bugzilla: 1714446
Acked-by: Andrew Jones <drjones@redhat.com>
Acked-by: Laszlo Ersek <lersek@redhat.com>

Commit 76b4ac28e975 introduced a regression, while trying to only clear
the DEBUG_VERBOSE bit, it also unconditionally sets other bits, normally
kept clear in the "silent" build.

Fix the macro by masking the cleared bits out.

Reported-by: Andrew Jones <drjones@redhat.com>
Suggested-by: Laszlo Ersek <lersek@redhat.com>
Signed-off-by: Philippe Mathieu-Daude <philmd@redhat.com>
---
 ArmVirtPkg/ArmVirtQemu.dsc       | 2 +-
 ArmVirtPkg/ArmVirtQemuKernel.dsc | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/ArmVirtPkg/ArmVirtQemu.dsc b/ArmVirtPkg/ArmVirtQemu.dsc
index aac4094..a44477f 100644
--- a/ArmVirtPkg/ArmVirtQemu.dsc
+++ b/ArmVirtPkg/ArmVirtQemu.dsc
@@ -423,7 +423,7 @@
   #
   OvmfPkg/QemuRamfbDxe/QemuRamfbDxe.inf {
     <PcdsFixedAtBuild>
-      gEfiMdePkgTokenSpaceGuid.PcdDebugPrintErrorLevel|0x8000004F
+      gEfiMdePkgTokenSpaceGuid.PcdDebugPrintErrorLevel|($(DEBUG_PRINT_ERROR_LEVEL)) & 0xFFBFFFFF
   }
   OvmfPkg/VirtioGpuDxe/VirtioGpu.inf
   OvmfPkg/PlatformDxe/Platform.inf
diff --git a/ArmVirtPkg/ArmVirtQemuKernel.dsc b/ArmVirtPkg/ArmVirtQemuKernel.dsc
index c9a635e..d28ac52 100644
--- a/ArmVirtPkg/ArmVirtQemuKernel.dsc
+++ b/ArmVirtPkg/ArmVirtQemuKernel.dsc
@@ -401,7 +401,7 @@
   #
   OvmfPkg/QemuRamfbDxe/QemuRamfbDxe.inf {
     <PcdsFixedAtBuild>
-      gEfiMdePkgTokenSpaceGuid.PcdDebugPrintErrorLevel|0x8000004F
+      gEfiMdePkgTokenSpaceGuid.PcdDebugPrintErrorLevel|($(DEBUG_PRINT_ERROR_LEVEL)) & 0xFFBFFFFF
   }
   OvmfPkg/VirtioGpuDxe/VirtioGpu.inf
   OvmfPkg/PlatformDxe/Platform.inf
-- 
1.8.3.1

